### PR TITLE
Add Image entities for vehicle status

### PR DESCRIPTION
**Pull request by WebSpider**
This adds 2 image entities that can display the vehicle status, as is done in the MySkoda App as well.
- One image is in 'Light' style
- One image is in 'Dark' style

The status is related to the window state, door state, bonnet state, etc.

Images are disabled by default, since it probably mostly is interesting to developers of custom cards.
They can be enabled manually in HA when going to the list of entities, and enabling them.

image

### CODE REVIEW

**Comment by WebSpider** [comment]
Fixes #342

**Comment by prvakt** [comment]
question 1: do we want to have those 2 under `SENSORS` because I think its more suitable for `DIAGNOSTICS`
question 2: I have noticed state for all images (also main render) is uknown.. I dont know if we want to handle state or not, but if yes maybe we can show something like available if image is present and unavailable if not?
```
    if is_available:
        self._state = "available"
    else:
        self._state = "unavailable"
    self.schedule_update_ha_state()
```
 
question 3: both images are same for my case, but from code I can see its retrieving correct path, so just want to check its same for you as well. (If it same, then does it make sense to have both?)
image
image

**Comment by prvakt** [comment]
question 4: is it updated with the periodical vehicle update?

**Comment by WebSpider** [comment]
> question 4: is it updated with the periodical vehicle update?

If there is an update to the state, the URL should change, triggering an update as part of the `ImageEntity`. Have not seen that happen yet, since my car is not close to my home at the moment.

**Comment by prvakt** [comment]
1 ) asked ChatGPT about the sensor vs diagnostic section and it suggest also sensors :)
```
Suggestion for Your Case:
Given that the image likely represents the state of doors and windows in real-time, I recommend placing it under Sensors, as it primarily visualizes the output of door/window state sensors. If, however, it includes additional diagnostic elements like system errors or advanced analytics, then Diagnostics would be a better fit.
```

2 ) the state of all image entities can be implemented later once we agree what to show. 

4 ) I'm running this version for hour with poll interval set to 15 minutes.. Initially the state was correct, then I have opened door but state was never changed, even I see periodical update of all sensors have been triggered, Status Render image was not updated..

**Comment by prvakt** [comment]
> The `ImageEntity` has some strange defaults. This should be corrected now.

Still doesnt work, I can see from logs that image url was changed from
```
https://mysmob.api.connect.skoda-auto.cz/api/v2/vehicle-status/render?carType=COMBI&vehicleState=1-1-1-1-0-0-0-0-0&lastModifiedAt=1733938917&dimension=3x&theme=LIGHT
```
to
```
https://mysmob.api.connect.skoda-auto.cz/api/v2/vehicle-status/render?carType=COMBI&vehicleState=1-1-3-1-0-0-0-0-0&lastModifiedAt=1733938917&dimension=3x&theme=LIGHT
```
but image was not updated and last updated timestamp was not changed either...

**Comment by WebSpider** [comment]
Thanks, will continue in the mysteries of the `ImageEntity` tomorrow 😂

**Comment by dvx76** [comment]
From https://developers.home-assistant.io/docs/core/entity/image/

>Implement async_image or image if your entity returns bytes of the image instead of providing a URL. Frontend will call async_image or image to fetch the image. If the image is fetched remotely, image data should be cached and the cache invalidated when image_last_updated is changed.

I would assume if the url is updated this would automatically invalidate the cache and force the url to be re-fetched. But maybe that's not happening?

File: custom_components/myskoda/image.py [hunk 1]
```diff
@@ -59,6 +64,38 @@ def __init__(
         super().__init__(coordinator, vin)
 
 
+class StatusImage(MySkodaImage):
+    """A render of the current status of the vehicle."""
+
+    _attr_should_poll: bool = True
+
+    async def _fetch_url(self, url: str) -> httpx.Response | None:
```
Comment (prvakt) [inline custom_components/myskoda/image.py:72]: `_fetch_url()` is not called from anywhere.. is this maybe the problem why status image is never updated?

Comment (WebSpider) [reply custom_components/myskoda/image.py:72]: _fetch_url is used in the parent class

Comment (prvakt) [reply custom_components/myskoda/image.py:72]: ah OK.. fyi I have tried to put some more logs in image.py but both `_fetch_url()` and also `image_url()` are called only once and never after

**Comment by WebSpider** [comment]
> From https://developers.home-assistant.io/docs/core/entity/image/
> 
> > Implement async_image or image if your entity returns bytes of the image instead of providing a URL. Frontend will call async_image or image to fetch the image. If the image is fetched remotely, image data should be cached and the cache invalidated when image_last_updated is changed.
> 
> I would assume if the url is updated this would automatically invalidate the cache and force the url to be re-fetched. But maybe that's not happening?

Rereading that piece of documentation, I think they mean we should implement a "last changed" property ourselves, so did that in last commit.

**Comment by prvakt** [comment]
> Rereading that piece of documentation, I think they mean we should implement a "last changed" property ourselves, so did that in last commit.

We are still missing something. "Last changed" was updated correctly, but image was not changed. I can see in MySkoda app that image has changed and door is now opened, and also new binary sensors which I have created today were updated accordingly but image was not

**Comment by WebSpider** [comment]
I will have a look at that parent class more thoroughly. This will not make v1.16.0.

File: custom_components/myskoda/image.py (hunk 1 above)
Comment (WebSpider) [reply custom_components/myskoda/image.py:70]: I have added cache invalidation code, lets see if this solves it.

**Comment by prvakt** [comment]
> I will have a look at that parent class more thoroughly. This will not make v1.16.0.

Good job! Can confirm it works as expected now ;)

**Review by prvakt — APPROVED** [review]

**Pull request merged**